apps/server/src/modules/tool/external-tool/uc/external-tool.uc.ts

```diff
@@ -88,8 +91,38 @@ export class ExternalToolUc {
 		return metadata;
 	}
 
-	private async ensurePermission(userId: EntityId, permission: Permission) {
+	private async ensurePermission(userId: EntityId, permission: Permission): Promise<void> {
 		const user: User = await this.authorizationService.getUserWithPermissions(userId);
 		this.authorizationService.checkAllPermissions(user, [permission]);
 	}
+
+	public async getDatasheet(userId: EntityId, externalToolId: EntityId): Promise<Buffer> {
+		const user: User = await this.authorizationService.getUserWithPermissions(userId);
+		this.authorizationService.checkAllPermissions(user, [Permission.TOOL_ADMIN]);
+		const dataSheetData: ExternalToolDatasheetTemplateData =
+			await this.externalToolService.getExternalToolDatasheetTemplateData(
+				externalToolId,
+				user.firstName,
+				user.lastName
+			);
+		const buffer: Promise<Buffer> = firstValueFrom(
+			this.pdfService.toBuffer('ExternalToolDatasheet', { locals: dataSheetData })
+		);
+
+		return buffer;
+	}
+
+	public async createDatasheetFilename(userId: EntityId, externalToolId: EntityId): Promise<string> {
+		const user: User = await this.authorizationService.getUserWithPermissions(userId);
+		const dataSheetData: ExternalToolDatasheetTemplateData =
+			await this.externalToolService.getExternalToolDatasheetTemplateData(
+				externalToolId,
+				user.firstName,
+				user.lastName
+			);
+
+		const fileName: string = this.externalToolService.createDatasheetFilename(dataSheetData);
```

Comment:
I doubt you need a whole service to concatenate a simple string. It does not have any reuse value. A private function in the uc would be enough.